DataFormats/CTPPSReco/interface/CTPPSLocalTrackLite.h

```diff
@@ -11,75 +11,150 @@
+    inline float getXUnc() const
     {
       return x_unc;
     }
 
     /// returns the vertical track position
-    float getY() const
+    inline float getY() const
     {
       return y;
     }
 
     /// returns the vertical track position uncertainty
-    float getYUnc() const
+    inline float getYUnc() const
     {
       return y_unc;
     }
 
     /// returns the track time
-    float getTime() const
+    inline float getTime() const
     {
       return time;
     }
 
     /// returns the track time uncertainty
-    float getTimeUnc() const
+    inline float getTimeUnc() const
     {
       return time_unc;
     }
 
+    /// returns the track horizontal angle
+    inline float getTx() const
+    {
+        return tx;
+    }
+
+    /// returns the track horizontal angle uncertainty
+    inline float getTxUnc() const
+    {
+        return tx_unc;
+    }
+    
+    /// returns the track vertical angle
+    inline float getTy() const
+    {
+        return ty;
+    }
+    
+    /// returns the track vertical angle uncertainty
+    inline float getTyUnc() const
+    {
+        return ty_unc;
+    }
+    
+    /// returns the track fit chi Squared over NDF
+    inline float getChiSquaredOverNDF() const
+    {
+        return chiSquaredOverNDF;
+    }
+
+    /// returns the track reconstruction info byte
+    inline unsigned int getRecoInfo() const
+    {
+        return (unsigned int)pixelTrack_reco_info;
+    }
+
+    /// returns the number of points used for fit
+    inline unsigned short getNumberOfPointsUsedForFit() const
+    {
+        return numberOfPointsUsedForFit;
+    }   
+
   protected:
     /// RP id
     uint32_t rpId;
 
+    /// local track parameterization
+    /// x = x0 + tx*(z-z0), y = y0 + ty*(z-z0)
+    /// x0, y0, z-z0 in mm
+    /// z0: position of the reference scoring plane (in the middle of the RP)
+
     /// horizontal hit position and uncertainty, mm
     float x, x_unc;
 
     /// vertical hit position and uncertainty, mm
     float y, y_unc;
 
+    /// horizontal angle and uncertainty, x = x0 + tx*(z-z0)
+    float tx, tx_unc;
+
+    /// vertical angle and uncertainty, y = y0 + ty*(z-z0)
+    float ty, ty_unc;
+
+    /// fit chi^2 over NDF
+    float chiSquaredOverNDF;
+
+    /// Track information byte for bx-shifted runs: 
+    /// pixelTrack_reco_info = notShiftedRun    -> Default value for tracks reconstructed in non-bx-shifted ROCs
+    /// pixelTrack_reco_info = allShiftedPlanes -> Track reconstructed in a bx-shifted ROC with bx-shifted planes only
+    /// pixelTrack_reco_info = noShiftedPlanes  -> Track reconstructed in a bx-shifted ROC with non-bx-shifted planes only
+    /// pixelTrack_reco_info = mixedPlanes      -> Track reconstructed in a bx-shifted ROC both with bx-shifted and non-bx-shifted planes
+    /// pixelTrack_reco_info = invalid          -> Dummy value. Assigned when pixelTrack_reco_info is not computed (i.e. non-pixel tracks)
+    ReconstructionInfo pixelTrack_reco_info;
```

Comment:
I was trying to Tree::Scan this directly in FWLite, but the result comes out somewhat random.
@Dr15Jones is there support for scoped enums for ROOT IO? do we need to create dictionaries for scoped enums?

Reply:
>is there support for scoped enums for ROOT IO? do we need to create dictionaries for scoped enums?

I don't know.

@pcanal

Reply:
BTW, the output of the method method ```unsigned int getRecoInfo()``` is displayed in a  correct range (from 0 to 5 here) with TTree::Scan.

Reply:
yes scope enums are supported and need a dictionary

Reply:
Here is an example

```
Events->SetAlias("pixelTrack_reco_info", "CTPPSLocalTrackLites_ctppsLocalTrackLiteProducer__reRECO.obj.pixelTrack_reco_info");
Events->SetAlias("getRecoInfo()", "CTPPSLocalTrackLites_ctppsLocalTrackLiteProducer__reRECO.obj.getRecoInfo()");

 Events->Scan("pixelTrack_reco_info:getRecoInfo()", "", "colsize=20")
*********************************************************************
*    Row   * Instance * pixelTrack_reco_info *        getRecoInfo() *
*********************************************************************
*        0 *        0 *               655365 *                    5 *
*        0 *        1 *               393216 *                    0 *
*        0 *        2 *               327680 *                    0 *
*        0 *        3 *               393216 *                    0 *
*        1 *        0 *                      *                      *
*        2 *        0 *               655365 *                    5 *
*        2 *        1 *                    5 *                    5 *
```

Reply:
> yes scope enums are supported and need a dictionary

@pcanal 
is there an example for this?

```<class ``` did not work
```<enum ``` compiled, but the results of the TTree::Scan did not change (and, perhaps not important here, it also looks like there is no version support)

Reply:
it looks like the main problem with correctly displaying in the TTree::Scan was the enum-base specification (``` enum class ReconstructionInfo: unsigned short {```).
I managed to get a working version with just this change, without the enum-base: https://github.com/cms-sw/cmssw/commit/81d550d30b8ec6ef9c2cf6e41c104ba12bd88594

@pcanal is the base type in enums supported?

Reply:
@slava77 You are right it is not (yet) supported.  Enums are assumed to be ints.

Reply:
What would be best for the code? Removing the base specification would increase the recoInfo variable size right? Is it necessary?